server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -639,4 +641,395 @@ private TrackingTool toTrackingTool(CRTrackingTool crTrackingTool) {
+            job.addProperty(new CRPropertyGenerator(prop.getName(), prop.getSrc(), prop.getXpath()));
+        }
+
+        for(Tab tab: jobConfig.getTabs()) {
+            job.addTab(new CRTab(tab.getName(), tab.getPath()));
+        }
+
+        for (ArtifactConfig artifactConfig: jobConfig.artifactConfigs()) {
+           job.addArtifact(artifactConfigToCRArtifact(artifactConfig));
+        }
+
+        if (jobConfig.isRunOnAllAgents()) {
+            job.setRunOnAllAgents(jobConfig.isRunOnAllAgents());
+        } else {
+            job.setRunInstanceCount(jobConfig.getRunInstanceCountValue());
+        }
+
+        for (Task task: jobConfig.tasks()) {
+           job.addTask(taskToCRTask(task));
+        }
+
+        job.setResources(jobConfig.resourceConfigs().resourceNames());
+        job.setElasticProfileId(jobConfig.getElasticProfileId());
+
+        if (jobConfig.getTimeout() != null)
+            job.setTimeout(Integer.valueOf(jobConfig.getTimeout()));
+
+        return job;
+    }
+
+    public CRTask taskToCRTask(Task task) {
+        if (task == null)
+            throw new ConfigConvertionException("task cannot be null");
+
+        if (task instanceof PluggableTask)
+            return pluggableTaskToCRPluggableTask((PluggableTask) task);
+        else if (task instanceof BuildTask) {
+            return buildTaskToCRBuildTask((BuildTask) task);
+        } else if (task instanceof ExecTask) {
+            return execTasktoCRExecTask((ExecTask) task);
+        } else if (task instanceof FetchTask) {
+            return fetchTaskToCRFetchTask((FetchTask) task);
+        } else if (task instanceof FetchPluggableArtifactTask) {
+            return fetchPluggableArtifactTaskToCRFetchPluggableTask((FetchPluggableArtifactTask) task);
+        } else
+            throw new RuntimeException(
+                    String.format("unknown type of task '%s'", task));
+    }
+
+    public CRFetchPluggableArtifactTask fetchPluggableArtifactTaskToCRFetchPluggableTask(FetchPluggableArtifactTask task) {
+        List<CRConfigurationProperty> configuration = configurationToCRConfiguration(task.getConfiguration());
+        CRFetchPluggableArtifactTask crTask = new CRFetchPluggableArtifactTask(
+                task.getStage().toString(),
+                task.getJob().toString(), task.getArtifactId(), configuration);
+        crTask.setPipelineName(task.getPipelineName().toString());
+        commonCRTaskMembers(crTask, task);
+        return crTask;
+    }
+
+    public CRFetchArtifactTask fetchTaskToCRFetchTask(FetchTask task) {
+        CRFetchArtifactTask fetchTask = new CRFetchArtifactTask(
+                task.getStage().toString(),
+                task.getJob().toString(),
+                task.getSrc());
+
+        fetchTask.setDestination(task.getDest());
+        fetchTask.setPipelineName(task.getPipelineName().toString());
+        fetchTask.setSourceIsDirectory(task.isSourceAFile());
```

Comment:
This _seems_ wrong.

Reply:
How so?

Reply:
I was looking mostly at the names, but shoudn't it be:

```
fetchTask.setSourceIsDirectory(!task.isSourceAFile());
```
?

The parameter's name is `srcIsDirectory`.

Reply:
Should be `fetchTask.setSourceIsDirectory(!task.isSourceAFile());`

Reply:
👍